### PR TITLE
Replace v1 aggregated_metrics API with faster v2 queries API in Fiddler Agent

### PR DESCRIPTION
**Pull request by bhalder**
## What does this PR do?

This PR rewrites Fiddler's Agent to use new faster v2 APIs. The PR ensures that the results published to DataDog continue to be backwards compatible.

### What inspired you to submit this pull request?

**Functional Reasons:**
- The existing implementation of Fiddler-Datadog integration uses v1 APIs - list_projects, and aggregated_metrics
- These APIs are planned to be deprecated
- These APIs are replaced by v2 APIs which have been in production for 6+ months. These APIs make much more efficient use of caching at every layer, and are faster.

**Maintainability Reasons:**
- We also remove hard dependency on Fiddler-Client, and instead uses requests directly
- This helps ensure that any fiddler-client dependency (e.g. pydantic) is not in conflict with any deployment environment
- We also wanted to add more tests, and replace "integration" like tests, with true unit tests

Screen Shot 2023-12-19 at 6 06 23 PM

## Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [x] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

## Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: fiddler/CHANGELOG.md
Comment (anawishnoff) [inline fiddler/CHANGELOG.md:7]:
```suggestion
* Updated to use faster v2 queries API.
```

**Review by anawishnoff — APPROVED** [review]
Looks good from the product side, other than the one change

**Comment by bhalder** [comment]
> Looks good from the product side, other than the one change

Thanks for the review! Committed the suggestion.

**Comment by bhalder** [comment]
@nmuesch can you help guide about who else needs to review this PR before merge?

File: fiddler/metadata.csv
Comment (iliakur) [inline fiddler/metadata.csv:19]: Shouldn't we consider this a breaking change if we're renaming metrics? Imagine someone who has a dashboard or monitor based on this metric? It's fine to make the change, but then we should bump the version to 2.0.0.

An alternative (which lets us keep 1.1.0) is to keep submitting metrics with the old wrong names **in addition to the correct names** and add a note to their descriptions saying they are deprecated give the names of the metrics with correct names as equivalents.

File: fiddler/tests/test_fiddler.py
Comment (iliakur) [inline fiddler/tests/test_fiddler.py:10]: Imports from `conftest` aren't the conventional way to use that file. Could you turn your mocks into `pytest.fixture`s and use them that way please?

**Review by iliakur — CHANGES_REQUESTED** [review]

File: fiddler/tests/test_fiddler.py
Comment (bhalder) [reply fiddler/tests/test_fiddler.py]: Great point! Have fixed it.
Thanks for the review 🙏

File: fiddler/metadata.csv
Comment (bhalder) [reply fiddler/metadata.csv:19]: Gotcha! Good to know that metric name updates require major version changes.
Have updated the version to be 2.0.0.
Thanks for the review! 🙏

**Comment by bhalder** [comment]
@nmuesch do you have any review comments on this PR? :)

**Review by nmuesch — APPROVED** [review]
Sorry for my delay; this all seems fine from my team's perspective. I'll defer to @iliakur for the final approval/merge requirements.

**Comment by bhalder** [comment]
Thanks a lot Nicholas! 🙏 
 I do see a new CI failure on Validate repository with python3.11 after rebase - will make fixes if applicable

**Comment by FlorentClarret** [comment]
> Thanks a lot Nicholas! 🙏 I do see a new CI failure on Validate repository with python3.11 after rebase, is it something to be looked into? I ran the checks locally - and the message is vague
> 
> The files are autogenerated without anything telling us clearly what exactly went wrong.
> 
> ```
> Validating data models for 1 checks ...
> fiddler:
>     Valid spec: __init__.py
>     Valid spec: defaults.py
> 
> Files with errors: 2
> Files valid: 3```
>  
> ```

We are releasing a new version of ddev to fix this. Nothing to do on your side, sorry for the inconvenience.

**Comment by bhalder** [comment]
Thanks @FlorentClarret !
Then it should be safe to merge this PR.

Who can help merge this?

**Pull request merged**
